### DISCUSSION TITLE
React to more ITV Hub changes

### DISCUSSION THREAD

**Discussion by skovatch** [Ascoware/get-iplayer-automator#381] [Announcements]
- itv: Fix parsing episodes from show pages on ITV Hub. Those pages also changed slightly, and the assumption that episode and series URLs had the same base is no longer valid.

<hr /><em>This discussion was created from the release <a href='https://github.com/Ascoware/get-iplayer-automator/releases/tag/v1.21.15'>React to more ITV Hub changes</a>.</em>

**Comment by SOLOSAIR** [comment]
Now picking up itv programmes, thanks.  I have an issue with my series link for 'Tonight', however. Where before it would pick up the 30 minute investigative series, now GiA is listing news bulletins and 'Peston' and listing them all as 'Tonight' 'added by series link' too. Not a huge problem, I can download all of the many episodes and delete all the unwanted content upon playing each to see which to delete but there are several every day so obviously it's just a nuisance. Strange it should do this.

**Reply by skovatch** [reply]
I would suggest renaming your current itv.cache file to something like itv.cache.bak, then re-run GiA so it can rebuild the cache. It sounds like in one of the updates the show search wasn't working correctly, so your cache now has incorrect entries.

The file is in "~/Library/Application Support/Get iPlayer Automator/"

Can you upload your current cache file so I can take a look at it?

**Comment by SOLOSAIR** [comment]
It seems the GitHub website won't let me upload a cache file.
I tried deleting the last broadcast (already watched) Tonight programme from history, searched in PVR and put it in series-link again.
When it downloaded, it no longer had 'series 24' in the show title upon download and the file appears in a 'Tonight' folder, rather than the previously created Series 24 folder - see screenshots. The Tonight series link is still downloading news bulletins and Peston programmes even after I renamed the cache to the name you suggested - see screenshots.
EDIT:  Just noticed another cache file - is this a problem? (screen shot added)
<img width="524" alt="Screenshot 2022-02-09 at 20 29 05" src="https://user-images.githubusercontent.com/37366634/153287530-9c80c153-3c68-4396-b7da-4ab85f4a0714.png">
<img width="792" alt="Screenshot 2022-02-09 at 20 33 03" src="https://user-images.githubusercontent.com/37366634/153287538-84129033-0bae-43d4-8ca2-7f74205a6d52.png">
<img width="514" alt="Screenshot 2022-02-09 at 20 39 44" src="https://user-images.githubusercontent.com/37366634/153287541-a444f11a-2e78-44a6-bf38-4782c0305dce.png">
<img width="110" alt="Screenshot 2022-02-09 at 20 57 43" src="https://user-images.githubusercontent.com/37366634/153289182-15cdfd85-9c7c-4f2c-acf7-1e5e94ece0ff.png">

**Comment by SOLOSAIR** [comment]
It didn't pick up my Paul O'Grady series link either. Upon searching in PVR Manager, there are no SERIES details so I see that it looks like I have to add all my PVR programmes again.
<img width="519" alt="Screenshot 2022-02-09 at 21 36 40" src="https://user-images.githubusercontent.com/37366634/153294973-b7ee75d8-2997-4aa4-ab76-78c823ac49de.png">
 details - see screenshot.

**Reply by skovatch** [reply]
Can you post a screenshot of the PVR dialog?

**Reply by SOLOSAIR** [reply]
![image](https://user-images.githubusercontent.com/37366634/153299142-50c1c1b5-6922-430e-82a2-2712f3829ee9.jpeg)
Programmes that pvr missed before are being listed in download queue upon refreshing cache - AFTER I first deleted the previous pvr titles that had series numbers on and re-added programmes from search.

I think the Tonight Series 24 will now add ok but my problem still lies with the unwanted news/peston pvr adds.

**Comment by skovatch** [comment]
OK, I think I figured out what's going on here. I wasn't completely correct in parsing series information from the ITV program pages. This should be fixed in the next update.
